### PR TITLE
add detail to arc1.rs comment and hint message

### LINKED ISSUE #250: arc1 seems like a giant step
There are several new concepts in the `arc1.rs` exercise, and I wonder if it would be useful to have a few gentle lead ins before this exercise.

In particular threads are a big leap in complexity with some new syntax (closures).

It would also be helpful to know what the arc1 code is trying to do in the first place. Reverse engineering the logic (with known missing lines of code) and figuring out threads and figuring out Arc, seems like too much for one exercise.

### PR DESCRIPTION
**Pull request by bmacer**
Adds an explanation to exercises/standard_library_types/arc1.rs to hopefully alleviate some confusion around this exercise.  Hopefully sufficient to close #250.

### CODE REVIEW

File: exercises/standard_library_types/arc1.rs [hunk 1]
```diff
@@ -1,7 +1,21 @@
 // arc1.rs
+// In this exercise, we are given Vec of u32 called "numbers" with values ranging
+// from 0 to 99 -- [ 0, 1, 2, ..., 98, 99 ]
+// We would like to use this set of numbers within 8 different threads simultaneously.
+// Each thread is going to get the sum of every eighth value, with an offset.
+// The first thread (offset 0), will sum 0, 8, 16, ...
+// The second thread (offset 1), will sum 1, 9, 17, ...
+// The third thread (offset 2}, will sum 2, 10, 18, ...
```
Comment (fmoko) [inline exercises/standard_library_types/arc1.rs:8]:
```suggestion
// The third thread (offset 2), will sum 2, 10, 18, ...
```

File: exercises/standard_library_types/arc1.rs [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+// ...
+// The eighth thread {offset 7}, will sum 7, 15, 23, ...
```
Comment (fmoko) [inline exercises/standard_library_types/arc1.rs:10]:
```suggestion
// The eighth thread (offset 7), will sum 7, 15, 23, ...
```

File: exercises/standard_library_types/arc1.rs [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+
+// Because we are using threads, our values need to be thread-safe.  Therefore,
+// we are using Arc.  We need to make a change in each of the two TODOs.
+
+
 // Make this code compile by filling in a value for `shared_numbers` where the
-// TODO comment is and create an initial binding for `child_numbers`
-// somewhere. Try not to create any copies of the `numbers` Vec!
+// first TODO comment is and create an initial binding for `child_numbers`
```
Comment (fmoko) [inline exercises/standard_library_types/arc1.rs:17]:
```suggestion
// first TODO comment is, and create an initial binding for `child_numbers`
```

**Review by fmoko — COMMENTED** [review]
Thank you for this! I think this is a great step in the right direction :) I added some stylistic pointers, and I'll have someone else look over it before we merge it, but this looks really good to me.

File: exercises/standard_library_types/arc1.rs [hunk 4]
```diff
@@ -1,7 +1,21 @@
 // arc1.rs
+// In this exercise, we are given Vec of u32 called "numbers" with values ranging
```
Comment (jrvidal) [inline exercises/standard_library_types/arc1.rs:2]:
```suggestion
// In this exercise, we are given a Vec of u32 called "numbers" with values ranging
```

**Review by jrvidal — APPROVED** [review]
LGTM, just a minor nit

**Comment by fmoko** [comment]
@all-contributors please add @bmacer for content

**Pull request merged**